Library/Homebrew/utils/shell_env.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+set +o posix
+
+KeepKeys=(
+  BASH
+  BASH_ARGC
+  BASH_ARGV
+  BREW_FILE_DIRECTORY
+  USR_LOCAL_BREW_FILE_DIRECTORY
+  USR_LOCAL_HOMEBREW_REPOSITORY
+  HOMEBREW_*
+  HOME
+  PWD
+  LANG
+  PATH
+  TERM
+)
+
+prune-shell-env() {
+  file=/tmp/homebrew-env
+
+  IFS="="
```

Comment:
I think this could be simplified quite a bit by doing something like `env | grep -E '^(HOMEBREW.*|HOME|PATH)=.*$'` and just writing `env` output to the original file. Thoughts?